functions.photon.php

```diff
@@ -220,7 +220,18 @@ function jetpack_photon_url( $image_url, $args = array(), $scheme = null ) {
 
 	return jetpack_photon_url_scheme( $photon_url, $scheme );
 }
-add_filter( 'jetpack_photon_url', 'jetpack_photon_url', 10, 3 );
+
+/**
+ * Add an easy way to photon-ize a URL that is safe to call even if Jetpack isn't active.
+ *
+ * See: https://jetpack.com/2013/07/11/photon-and-themes/
+ */
+function add_jetpack_photon_url_filter() {
+	if ( Jetpack::is_module_active( 'photon' ) ) {
```

Comment:
One caveat here is that the Tiled Gallery module uses Photon even if it's not enabled. I have tried using Tiled Galleries in the block editor, and this change breaks them in a subtle way. With Photon enabled, the tiled mosaic mode works well, but when you disable Photon, it looks like the sizes get calculated incorrectly, and the broken gallery doesn't get fixed even when you enable Photon. Only editing the gallery helps afterwards.
Perhaps we can just set this filter here unconditionally to avoid breaking existing behavior?

Reply:
By intent, Photon needs to be able to be used without the module being active, so we can probably keep the `plugins_loaded` change but need to remove the module check.

Reply:
Thanks for bringing this up; I didn’t realize that the Tiled Gallery could be affected! I added it to the testing instructions.

I haven’t been able to reproduce this problem. The tiled galleries in mosaic mode look fine on my test site, regardless of whether the Photon/Image CDN/Site Accelerator 😄module is activated or deactivated. Are you seeing the problem in the post editor in the post, or both?

It looks like the intention is that the `jetpack_photon_url` filter should be used only when the [Photon module is active](https://github.com/Automattic/jetpack/blob/master/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php#L24). So maybe we should determine the cause of the broken gallery and change the Tiled Gallery module. For example, the `jetpack_photon_url()` method can be called directly, like it is [here](https://github.com/Automattic/jetpack/blob/master/modules/tiled-gallery/tiled-gallery/tiled-gallery-item.php#L37), without going through the`jetpack_photon_url` filter.

Reply:
Hmm, that's weird, I can't reproduce the problem today using the same steps. @kraftbj can you think of any reason why we need to keep the filter on if Accelerator is not loaded? The function that converts a URL to a CDN URL seems to be working fine for Tiled Galleries.

Reply:
I can see at least one place where we're using the filter instead of the direct function call: VideoPress. Should we just use the function directly in places like these and remove the filter when Accelerator is enabled?

Reply:
Since the module check could change existing behavior, I’ll remove it.